src/jit/flowgraph.cpp

```diff
@@ -23870,6 +23888,253 @@ void Compiler::fgUpdateFinallyTargetFlags()
             }
         }
     }
-
 #endif // FEATURE_EH_FUNCLETS && defined(_TARGET_ARM_)
 }
+
+class FatCalliTransformer
+{
+public:
+    FatCalliTransformer(Compiler* compiler) : compiler(compiler)
+    {
+    }
+
+    void Run()
+    {
+        BasicBlock* firstBasicBlock = compiler->fgFirstBB;
+        for (BasicBlock* block = firstBasicBlock; block != nullptr; block = block->bbNext)
+        {
+            TransformBlock(block);
+        }
+    }
+
+private:
+    void TransformBlock(BasicBlock* block)
+    {
+        for (GenTreeStmt* stmt = block->firstStmt(); stmt != nullptr; stmt = stmt->gtNextStmt)
+        {
+            if (ContainsFatCalli(stmt))
+            {
+                StatementTransformer stmtTransformer(compiler, block, stmt);
+                stmtTransformer.Run();
+            }
+        }
+    }
+
+    bool ContainsFatCalli(GenTreeStmt* stmt)
+    {
+        GenTreePtr fatPointerCandidate = stmt->gtStmtExpr;
+        if (fatPointerCandidate->OperIsAssignment())
+        {
+            fatPointerCandidate = fatPointerCandidate->gtGetOp2();
+        }
+        return fatPointerCandidate->IsCall() && fatPointerCandidate->AsCall()->IsFatPointerCandidate();
+    }
+
```

Comment:
So the only "fat pointer candidate" forms are statements of the form: `call()` or `var = call()`? Do the importer changes you make guarantee this? Is there a reason why we can't just handle `call` nodes anywhere in the tree?